### ISSUE TITLE
Latest Open3D causes issues while importing chamfer distance package.

### ISSUE THREAD

**Issue by adityamwagh** [ZrrSkywalker/Point-M2AE#4]
Hi, I have been trying to run your code, and I have found that `pip install -r requirements.txt` installs the latest Open3D package form PyPI and also installs torch v 1.13.0 from PyPI as a dependency. 

`conda install pytorch torchvision cudatoolkit` also installs PyTorch, but the package is called `pytorch` and not `torch`. So there are two versions of PyTorch installed simultaneously in the environment.

This causes issues while executing the instruction `import chamfer`. Furthermore, he GCC version needs to be <=10 to successfully compile the extensions.

Can you update the README to reflect this? It would be great if you could provide your conda `environment.yml` file with which you were successfully able to run your code.

**Comment by ZiyuGuo99** [comment]
Thanks for your kind suggestion.
Open3D's version is supposed to be open3d==0.9, which I have updated in the repo.

Here is my environment.yaml:

name: pointm2ae
channels:
  - pytorch
  - https://mirrors.sjtug.sjtu.edu.cn/anaconda/pkgs/free
  - https://mirrors.sjtug.sjtu.edu.cn/anaconda/cloud/conda-forge
  - defaults
dependencies:
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=1_gnu
  - blas=1.0=mkl
  - bzip2=1.0.8=h7f98852_4
  - ca-certificates=2021.5.30=ha878542_0
  - certifi=2021.5.30=py37h89c1867_0
  - cudatoolkit=11.0.3=h15472ef_8
  - ffmpeg=4.3=hf484d3e_0
  - freetype=2.10.4=h0708190_1
  - gmp=6.2.1=h58526e2_0
  - gnutls=3.6.13=h85f3911_1
  - intel-openmp=2021.3.0=h06a4308_3350
  - jpeg=9b=0
  - lame=3.100=h7f98852_1001
  - lcms2=2.12=h3be6417_0
  - ld_impl_linux-64=2.36.1=hea4e1c9_2
  - libffi=3.3=h58526e2_2
  - libgcc-ng=11.1.0=hc902ee8_8
  - libgomp=11.1.0=hc902ee8_8
  - libiconv=1.16=h516909a_0
  - libpng=1.6.37=h21135ba_2
  - libstdcxx-ng=11.1.0=h56837e0_8
  - libtiff=4.2.0=h85742a9_0
  - libuv=1.42.0=h7f98852_0
  - libwebp-base=1.2.0=h7f98852_2
  - lz4-c=1.9.3=h9c3ff4c_1
  - mkl=2021.3.0=h06a4308_520
  - mkl-service=2.4.0=py37h5e8e339_0
  - mkl_fft=1.3.0=py37h42c9631_2
  - mkl_random=1.2.2=py37h219a48f_0
  - ncurses=6.2=h58526e2_4
  - nettle=3.6=he412f7d_0
  - numpy=1.20.3=py37hf144106_0
  - numpy-base=1.20.3=py37h74d4b33_0
  - olefile=0.46=pyh9f0ad1d_1
  - openh264=2.1.1=h780b84a_0
  - openjpeg=2.4.0=hb52868f_1
  - openssl=1.1.1k=h7f98852_0
  - pip=21.2.2=pyhd8ed1ab_0
  - python=3.7.10=hffdb5ce_100_cpython
  - python_abi=3.7=2_cp37m
  - pytorch=1.7.0=py3.7_cuda11.0.221_cudnn8.0.3_0
  - readline=8.1=h46c0cb4_0
  - setuptools=49.6.0=py37h89c1867_3
  - six=1.16.0=pyh6c4a22f_0
  - sqlite=3.36.0=h9cd32fc_0
  - tk=8.6.10=h21135ba_1
  - torchaudio=0.7.0=py37
  - torchvision=0.8.0=py37_cu110
  - typing_extensions=3.10.0.0=pyha770c72_0
  - wheel=0.36.2=pyhd3deb0d_0
  - xz=5.2.5=h516909a_1
  - zlib=1.2.11=h516909a_1010
  - zstd=1.4.9=ha95c52a_0
  - pip:
    - apex==0.1
    - argon2-cffi==21.3.0
    - argon2-cffi-bindings==21.2.0
    - argparse==1.4.0
    - attrs==21.2.0
    - backcall==0.2.0
    - beautifulsoup4==4.11.1
    - bleach==5.0.0
    - blobfile==1.2.3
    - cached-property==1.5.2
    - cffi==1.15.0
    - charset-normalizer==2.0.4
    - cycler==0.11.0
    - dataclasses==0.6
    - debugpy==1.6.0
    - decorator==5.1.1
    - deepspeed==0.4.0
    - defusedxml==0.7.1
    - easydict==1.9
    - einops==0.3.0
    - entrypoints==0.4
    - fastjsonschema==2.15.3
    - filelock==3.0.12
    - fonttools==4.33.3
    - ftfy==6.1.1
    - future==0.18.2
    - gdown==4.5.1
    - h5py==3.3.0
    - idna==3.2
    - imageio==2.25.0
    - importlib-metadata==4.6.3
    - importlib-resources==5.7.1
    - iniconfig==1.1.1
    - ipykernel==6.13.0
    - ipython==7.33.0
    - ipython-genutils==0.2.0
    - ipywidgets==7.7.0
    - jedi==0.18.1
    - jinja2==3.1.2
    - joblib==1.1.0
    - jsonschema==4.5.1
    - jupyter-client==7.3.1
    - jupyter-core==4.10.0
    - jupyterlab-pygments==0.2.2
    - jupyterlab-widgets==1.1.0
    - kiwisolver==1.4.2
    - knn-cuda==0.2
    - llvmlite==0.36.0
    - logger==1.4
    - markupsafe==2.1.1
    - matplotlib==3.5.2
    - matplotlib-inline==0.1.3
    - mistune==0.8.4
    - mypy==0.910
    - mypy-extensions==0.4.3
    - nbclient==0.6.3
    - nbconvert==6.5.0
    - nbformat==5.4.0
    - nest-asyncio==1.5.5
    - networkx==2.2
    - ninja==1.10.2
    - notebook==6.4.11
    - numba==0.53.1
    - open3d==0.9.0.0
    - opencv-python==4.5.5.64
    - packaging==21.0
    - pandocfilters==1.5.0
    - parso==0.8.3
    - pexpect==4.8.0
    - pickleshare==0.7.5
    - pillow==9.4.0
    - pluggy==0.13.1
    - plyfile==0.7.4
    - pointnet2==0.0.0
    - pointnet2-ops==3.0.0
    - prometheus-client==0.14.1
    - prompt-toolkit==3.0.29
    - protobuf==3.17.3
    - psutil==5.8.0
    - ptyprocess==0.7.0
    - py==1.10.0
    - pycparser==2.21
    - pycryptodomex==3.10.1
    - pygments==2.12.0
    - pyparsing==2.4.7
    - pyrsistent==0.18.1
    - pysocks==1.7.1
    - pytest==6.2.4
    - python-dateutil==2.8.2
    - pyyaml==5.4.1
    - pyzmq==22.3.0
    - regex==2022.9.13
    - requests==2.26.0
    - scikit-learn==1.0.2
    - scipy==1.7.1
    - send2trash==1.8.0
    - soupsieve==2.3.2.post1
    - tensorboardx==1.8
    - termcolor==1.1.0
    - terminado==0.13.3
    - threadpoolctl==3.1.0
    - timm==0.4.5
    - tinycss2==1.1.1
    - toml==0.10.2
    - torch-scatter==2.0.7
    - tornado==6.1
    - tqdm==4.62.0
    - traitlets==5.1.1
    - transforms3d==0.3.1
    - trimesh==2.35.39
    - triton==1.0.0
    - typed-ast==1.4.3
    - urllib3==1.26.6
    - wcwidth==0.2.5
    - webencodings==0.5.1
    - widgetsnbextension==3.6.0
    - xmltodict==0.12.0
    - zipp==3.5.0
prefix: /home/miniconda3/envs/pointm2ae

**Comment by Dario-Mantegazza** [comment]
I will piggyback this issue since my question is related. 
You say that open3d is 0.9.0 but that version is compatible only with python 3.6 at max. Also from the dump above it seems that you are using python 3.7 while in the repo readme you set it to 3.8. 
Since I'm in a rush I will try and use your code with the latest open3d and python3.8 but in the meantime I kindly ask you to fix this discrepancy.
Thanks for the great work and otherwise excellent repo :)

**Comment by adityamwagh** [comment]
@ZrrSkywalker It would be quite useful to make a docker container with the correct configuration.

**Comment by Dario-Mantegazza** [comment]
I'm interested in adapting your code to run only the features extraction like you do for the modelnet40 SVM test.
Boy I had to jump through stuff to manage to make the code run; in fairness, I'm limited to running the code in a Apptainer/Singularity container, but still the install instructions provided here are half of the process.

In case you are interested we could work a bit together and release a singularity definition file that is somewhat compatible with the code.
